tests/backend/test_deploy.py

```diff
@@ -0,0 +1,297 @@
+        return {"status": "ok", "buildId": "123"}
+
+
+    @app.get("/deploy")
+    def get_status(
+        state: Annotated[State, Depends(lambda: state)],
+        authorization: Annotated[str, Header(description="The API key")],
+    ):
+
+        def get_time(n):
+            return (datetime.now(pytz.timezone('UTC')) + timedelta(seconds=n)).isoformat()
+
+        state.log_counter += 1
+        if (authorization == "Bearer full"):
+            if state.log_counter == 1: # first call is to checking if app exist
+                return {
+                    "logs": [],
+                    "status": {
+                        "url": None,
+                        "status": "PENDING",
+                    }
+                }
+            if state.log_counter == 2:
+                return {
+                    "logs": [
+                        {"log": f"{get_time(-7)} stdout F <envs>{state.envs}</envs>"},
+                        {"log": f"{get_time(-6)} stdout F <log0/>"},
+                        {"log": f"{get_time(-5)} stdout F <log1/>"},
+                    ],
+                    "status": {
+                        "url": None,
+                        "status": "BUILDING",
+                    }
+                }
+            if state.log_counter == 3:
+                return {
+                    "logs": [
+                        {"log": f"{get_time(-2)} stdout F <log3/>"},
+                        {"log": f"{get_time(-4)} stdout F <log2/>"},
+                    ],
+                    "status": {
+                        "url": "https://full.my-app.com",
+                        "status": "COMPLETED",
+                    }
+                }
+        if (authorization == "Bearer test"):
+            return {
+                "logs": [
+                    {"log": f"20210813163223 stdout F <envs>{state.envs}</envs>"},
+                ],
+                "status": {
+                    "url": "https://my-app.com",
+                    "status": "COMPLETED",
+                }
+            }
+        return {
+            "logs": [],
+            "status": {
+                "url": None,
+                "status": "FAILED",
+            }
+        }
+
+    @app.delete("/deploy")
+    def undeploy(
+        authorization: Annotated[str, Header(description="The API key")],
+    ):
+        return {"status": "ok"}
+    return app
+
+
+class Server(uvicorn.Server):
+    def __init__(self):
+        config = uvicorn.Config(create_app(), host="127.0.0.1", port=8888, log_level="info")
+        super().__init__(config)
+        self.keep_running = True
+
+    def install_signal_handlers(self):
+        pass
+
+    @contextlib.contextmanager
+    def run_in_thread(self):
+        thread = threading.Thread(target=self.run)
+        thread.start()
+        try:
+            while not self.started:
+                time.sleep(1e-3)
+            yield
+        finally:
+            self.should_exit = True
+            thread.join()
```

Comment:
I would advocate to move this part into `tests/backend/fixtures/cloud_deploy_fixture.py`. It will improve the readability of tests